### ISSUE TITLE
ICE: encoding version problem with ice 3.5.1

### ISSUE THREAD

**Issue by indust** [mumble-voip/mumble#1252]
I get error while using murmur.ice 

Ice::UnsupportedEncodingException:  
protocol error: unsupported encoding version: 1.1  
(can only support encodings compatible with version 1.1) 

http://doc.zeroc.com/pages/viewpage.action?pageId=2523138
It seem that Ice 3.5 uses the new 1.1 encoding by default , so how can i solve it?

**Comment by sapien2** [comment]
This issue is first search result for described problem. Can anyone explain how to configure existing murmur installation with ice 3.4 to solve this issue?

**Comment by sjberry** [comment]
@Kissaki I ran into this issue recently when working with the newest stable release of the Murmur server (1.2.10) and zeroc-ice 3.6 for Python. The installation process for Ice in Python was convoluted before 3.6 as it relied on binary distributions rather than pypi. It seems that ZeroC has addressed this since it is now possible to work with Ice and IcePy in Python virtual environments.

That said, the only version available on pypi is 3.6 which uses the 1.1 protocol by default. I did not test the development snapshot of Murmur (1.3.0), but 1.2.10 uses the 1.0 protocol. This isn't a problem necessarily as zeroc-ice also supports the 1.0 protocol. It's just not the default setting.

Check this page https://doc.zeroc.com/display/Ice36/Protocol+and+Encoding+Versions for a little bit more detail here. To address this in your Python script ([mice.py](https://github.com/mumble-voip/mumble-scripts/blob/master/Helpers/mice.py)) you can add the line:

```python
props.setProperty('Ice.Default.EncodingVersion', '1.0')
```

Around [mice.py#L61](https://github.com/mumble-voip/mumble-scripts/blob/master/Helpers/mice.py#L61) where you're otherwise setting properties. You'd most certainly be able to do this in whatever other language you're using to interface with the slice.

In short, even though Murmur only supports encoding version 1.0, you're not restricted to an older version of ZeroC's Ice just to use it. This is probably why they didn't bump to a major version of ZeroC; backwards compatibility appears to be in check.

Of course this may be a moot point come 1.3.0 as the standard release as you all might (?) be switching to protocol 1.1.

Hope this helps anyone who stumbles across this issue in their searches.

**Comment by tjhorner** [comment]
Getting the same issue with `ice@3.6.1` for Node.js.

**Comment by mkrautz** [comment]
It's worth noting that we don't guarantee a particular version of the Ice protocol.

Our own binaries of Mumble 1.2.x use Ice 3.4, while most 1.3.x binaries use Ice 3.5 right now (except OS X).

On Linux, it's harder, because we use the system's Ice installation, and that's mostly out of our hands.

One thing we could do is to put Ice.Default.EncodingVersion = 1.0 into our default murmur.ini and simply mandate that version.

But as @sjberry reports, it seems like clients don't themselves fall back to 1.0? If so... that's kind of a usability problem.

(But so is Ice in the first place, frankly.)

**Comment by Krzmbrzl** [comment]
I think there's not much more we can do on this. If this problem arises again, please comment and let me know. Then I'll reopen this issue
